src/test/kotlin/no/nav/syfo/aktivitetskrav/api/AktivitetskravApiSpek.kt

```diff
@@ -231,7 +231,138 @@ class AktivitetskravApiSpek : Spek({
+                        with(
+                            handleRequest(HttpMethod.Post, urlVurderAktivitetskrav) {
+                                addHeader(HttpHeaders.ContentType, ContentType.Application.Json.toString())
+                                addHeader(HttpHeaders.Authorization, bearerHeader(validToken))
+                                addHeader(NAV_PERSONIDENT_HEADER, UserConstants.ARBEIDSTAKER_PERSONIDENT.value)
+                                setBody(objectMapper.writeValueAsString(vurderingOppfyltRequestDTO))
+                            }
+                        ) {
+                            response.status() shouldBeEqualTo HttpStatusCode.OK
+                            val producerRecordSlot = slot<ProducerRecord<String, KafkaAktivitetskravVurdering>>()
+                            verify(exactly = 1) {
+                                kafkaProducer.send(capture(producerRecordSlot))
+                            }
+
+                            val pAktivitetskravList = database.getAktivitetskrav(personIdent = UserConstants.ARBEIDSTAKER_PERSONIDENT)
+                            pAktivitetskravList.size shouldBeEqualTo 1
+                            val aktivitetskrav = pAktivitetskravList.first()
+                            aktivitetskrav.status shouldBeEqualTo vurderingOppfyltRequestDTO.status.name
+                            aktivitetskrav.stoppunktAt shouldBeEqualTo LocalDate.now()
+
+                            val pAktivitetskravVurderingList = database.getAktivitetskravVurderinger(aktivitetskravId = aktivitetskrav.id)
+                            pAktivitetskravVurderingList.size shouldBeEqualTo 1
+                            val aktivitetskravVurdering = pAktivitetskravVurderingList.first()
+
+                            val kafkaAktivitetskravVurdering = producerRecordSlot.captured.value()
+                            kafkaAktivitetskravVurdering.personIdent shouldBeEqualTo UserConstants.ARBEIDSTAKER_PERSONIDENT.value
+                            kafkaAktivitetskravVurdering.status shouldBeEqualTo aktivitetskrav.status
+                            kafkaAktivitetskravVurdering.beskrivelse shouldBeEqualTo "Aktivitetskravet er oppfylt"
+                            kafkaAktivitetskravVurdering.arsaker shouldBeEqualTo listOf(VurderingArsak.FRISKMELDT.name)
+                            kafkaAktivitetskravVurdering.updatedBy shouldBeEqualTo UserConstants.VEILEDER_IDENT
+                            kafkaAktivitetskravVurdering.sistVurdert?.truncatedTo(ChronoUnit.MILLIS) shouldBeEqualTo aktivitetskravVurdering.createdAt.truncatedTo(
+                                ChronoUnit.MILLIS
+                            )
+                            kafkaAktivitetskravVurdering.stoppunktAt shouldBeEqualTo aktivitetskrav.stoppunktAt
+                        }
+                    }
+                }
+                describe("Unhappy path") {
```

Comment:
Testene for headers og token er vel like overalt? Kanskje det hadde gått an å abstrahert ut noe og bare sendt inn en url? Eller er det kanskje dumt å "gjemme" noen tester? 🤔

Reply:
Ja, kan se om jeg kan få redusert litt duplisering der...